api/dataset/satellite.go

```diff
@@ -216,8 +235,10 @@ func (s *Satellite) CreateDataset(rootDataPath string, datasetName string, confi
 					d3mIDRunning = d3mIDRunning + 1
 					d3mIDs[groupID] = d3mID
 				}
+				// remove values that are not needed based on the headerNames (expects values, expectedHeaders and headerNames to be IN ORDER)
+				csvLine := removeMissingValues(expectedHeaders, headerNames, []string{fmt.Sprintf("%d", d3mID), path.Base(targetImageFilename), groupID, band, timestamp, coordinates.String(), label, coordinates.ToGeometryString()})
```

Comment:
given that every row will have the same transform applied, there should be a way to not duplicate effort quite so much